src/aktualizr_secondary/aktualizr_secondary.cc

```diff
@@ -113,14 +86,17 @@ bool AktualizrSecondary::putRootResp(const std::string& root, bool director) {
 }
 
 bool AktualizrSecondary::sendFirmwareResp(const std::shared_ptr<std::string>& firmware) {
-  if (target_ == nullptr) {
-    LOG_ERROR << "No valid installation target found";
+  auto targetsForThisEcu = director_repo_.getTargets(getSerial());
```

Comment:
Should we be rechecking the metadata (without fetching) like we do on the Primary before installation? It seems redundant, but it's a protection against unexpected changes in the time since the metadata was received.

Reply:
Yes, I agree, we should check at least an image file size (binary case) and its hash (for both ostree and binary).

Reply:
I think it's also worth rechecking the whole metadata chain from the metadata, but yeah, checking the object's hash/size is even more important.